src/components/NewGame/hooks/NewGameActions.ts

```diff
@@ -0,0 +1,183 @@
+import {
+  NewGameActionType,
+  PREDEFINED_RULES,
+  COLORS,
+  TEAMS,
+  RULES,
+  RulesPayload,
+  SUBMIT,
+  COMPLETE_STEP,
+  SETUP,
+} from './NewGameActionTypes';
+import { GameListItem, TeamColor } from '../NewGameTypes';
+
+export const setupAction = (value: string): NewGameActionType => {
+  return {
+    type: SETUP,
+    payload: value,
+  };
+};
+
+export const colorsAction = (teamColor: TeamColor, teamColors: TeamColor[], teams: number): NewGameActionType => {
+  if (teamColors.indexOf(teamColor) >= 0) {
+    const newTeamColors = teamColors.filter(color => color !== teamColor);
+    return { type: COLORS, payload: newTeamColors };
+  }
+
+  if (teamColors.length >= teams) {
+    return { type: COLORS, payload: teamColors };
+  }
+
+  const newTeamColors = [...teamColors, teamColor];
+  return { type: COLORS, payload: newTeamColors };
+};
+
+export const teamsAction = (colorsList: TeamColor[], teamColors: TeamColor[], newTeams: number): NewGameActionType => {
+  let newTeamColors = [...teamColors];
+
+  if (newTeams < teamColors.length) {
+    newTeamColors = teamColors.slice(0, newTeams);
+  }
+
+  if (newTeams > teamColors.length) {
+    const availableColors = colorsList.filter(teamColor => teamColors.indexOf(teamColor) === -1).reverse();
+
+    if (availableColors.length) {
+      newTeamColors = new Array(newTeams - teamColors.length)
+        .fill(true)
+        .reduce(totTeams => [...totTeams, availableColors.pop()], teamColors);
+    }
+  }
+
+  return {
+    type: TEAMS,
+    payload: {
+      teams: newTeams,
+      teamColors: newTeamColors,
+    },
+  };
+};
+
+export const predefinedRulesAction = (rules: GameListItem, teams: number): NewGameActionType => {
```

Comment:
When you are calling this here:
https://github.com/Online-Scoreboard/online-scoreboard/pull/31/files#diff-f662c003bace3d961ead65eb016805b4R279
first argument is of type `GameListItem`
which is:
```
export interface GameListItem {
  name: string;
  minTeamSize: number;
  maxTeamSize: number;
  startingScore: number;
  winningScore: number;
  winningScoreEnabled: boolean;
  scoringSystem: ScoringSystem;
  isMatchesBased: boolean;
}
```
Why are you casting `minTeamSize` and `maxTeamSize` to  `number` on lines 64 and 67 in this file?
Maybe I am missing something!

Reply:
yes, good point. Problem is that when those values are coming from the Redux store, they are already cast into numbers. However, when passed from the input fields by the user, they're still in a string format. The number casting is to make sure the values are always numbered whether they are coming from the input fields or from the redux store. But maybe I can refactor this logic and simplify that

Reply:
Hm, I am not familiar with the project structure so apologies if it comes off a bit ignorant, but assuming that the value from input field is handled by something in the backend/some control block, wouldn't casting it there once be easier to keep it consistent with how they are coming from the Redux store?

Reply:
oh no, the user input comes directly in here, then it goes to the backed